src/containers/track-page/store/sagas.js

```diff
@@ -140,39 +141,43 @@ function* getMoreByThisArtist(trackId, ownerHandle) {
 
 function* watchFetchTrack() {
   yield takeEvery(trackPageActions.FETCH_TRACK, function* (action) {
-    const { trackId, trackName, ownerHandle, canBeUnlisted } = action
-    const ids = canBeUnlisted
-      ? [{ id: trackId, url_title: trackName, handle: ownerHandle }]
-      : [trackId]
-
+    const { trackId, handle, slug, canBeUnlisted } = action
     try {
-      yield fork(getMoreByThisArtist, trackId, ownerHandle)
-      const trackIds = yield call(retrieveTracks, {
-        trackIds: ids,
-        canBeUnlisted,
-        withStems: true,
-        withRemixes: true,
-        withRemixParents: true
-      })
-      if (
-        !trackIds ||
-        !trackIds.length ||
-        trackIds.every(track => track === undefined || !track.track_id)
-      ) {
-        // If no tracks because no internet, do nothing. Else navigate to 404.
+      let track
+      if (!trackId) {
+        track = yield call(retrieveTrackByHandleAndSlug, {
+          handle,
+          slug
+        })
+      } else {
+        const ids = canBeUnlisted
+          ? [{ id: trackId, url_title: slug, handle }]
+          : [trackId]
+        const tracks = yield call(retrieveTracks, {
+          trackIds: ids,
+          canBeUnlisted,
+          withStems: true,
+          withRemixes: true,
+          withRemixParents: true
+        })
+        track = tracks && tracks.length === 1 ? tracks[0] : null
+      }
+      if (!track) {
         const isReachable = yield select(getIsReachable)
         if (isReachable) {
           yield put(pushRoute(NOT_FOUND_PAGE))
           return
         }
+      } else {
+        yield fork(getMoreByThisArtist, track.track_id, handle)
```

Comment:
Note that before this change we could start to get tracks by the artist because we had the track ID ahead of time.